File: GeneratedEntityReaderClasses/Components/IngestionComponent.cs
namespace GeneratedNoitaClasses;
public class IngestionComponent : NoitaComponentBase,  iNoitaType<IngestionComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt64 ingestion_size { get; set; } = new();
	public NoitaInt64 ingestion_capacity { get; set; } = new();
	public NoitaUInt ingestion_cooldown_delay_frames { get; set; } = new();
	public NoitaUInt ingestion_reduce_every_n_frame { get; set; } = new();
	public NoitaFloat overingestion_damage { get; set; } = new();
	public NoitaFloat blood_healing_speed { get; set; } = new();
	public NoitaString ingestion_satiation_material_tag { get; set; } = new();
	public NoitaInt m_ingestion_cooldown_frames { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		ingestion_size.Read(s);
		ingestion_capacity.Read(s);
		ingestion_cooldown_delay_frames.Read(s);
		ingestion_reduce_every_n_frame.Read(s);
		overingestion_damage.Read(s);
		blood_healing_speed.Read(s);
		ingestion_satiation_material_tag.Read(s);
		m_ingestion_cooldown_frames.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		ingestion_size.Write(s);
		ingestion_capacity.Write(s);
		ingestion_cooldown_delay_frames.Write(s);
		ingestion_reduce_every_n_frame.Write(s);
		overingestion_damage.Write(s);
		blood_healing_speed.Write(s);
		ingestion_satiation_material_tag.Write(s);
		m_ingestion_cooldown_frames.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/IngestionComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt64
from BasicNoitaTypes import NoitaUInt
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class IngestionComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    ingestion_size:NoitaInt64 = field(default_factory=NoitaInt64)
    ingestion_capacity:NoitaInt64 = field(default_factory=NoitaInt64)
    ingestion_cooldown_delay_frames:NoitaUInt = field(default_factory=NoitaUInt)
    ingestion_reduce_every_n_frame:NoitaUInt = field(default_factory=NoitaUInt)
    overingestion_damage:NoitaFloat = field(default_factory=NoitaFloat)
    blood_healing_speed:NoitaFloat = field(default_factory=NoitaFloat)
    ingestion_satiation_material_tag:NoitaString = field(default_factory=NoitaString)
    m_ingestion_cooldown_frames:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.ingestion_size.Read(s)
        self.ingestion_capacity.Read(s)
        self.ingestion_cooldown_delay_frames.Read(s)
        self.ingestion_reduce_every_n_frame.Read(s)
        self.overingestion_damage.Read(s)
        self.blood_healing_speed.Read(s)
        self.ingestion_satiation_material_tag.Read(s)
        self.m_ingestion_cooldown_frames.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.ingestion_size.Write(s)
        self.ingestion_capacity.Write(s)
        self.ingestion_cooldown_delay_frames.Write(s)
        self.ingestion_reduce_every_n_frame.Write(s)
        self.overingestion_damage.Write(s)
        self.blood_healing_speed.Write(s)
        self.ingestion_satiation_material_tag.Write(s)
        self.m_ingestion_cooldown_frames.Write(s)

